backend/webapp/src/main/java/org/bonitasoft/web/designer/controller/upload/WebResourceUploader.java

```diff
@@ -0,0 +1,101 @@
+import org.bonitasoft.web.designer.repository.exception.NotFoundException;
+import org.bonitasoft.web.designer.rest.ErrorMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.multipart.MultipartFile;
+
+public class WebResourceUploader<T extends WebResourceable> {
+
+    protected static final Logger logger = LoggerFactory.getLogger(WebResourceUploader.class);
+
+    private Repository<T> repository;
+    private WebResourceRepository<T> webResourceRepository;
+
+    public WebResourceUploader(Repository<T> repository, WebResourceRepository<T> webResourceRepository) {
+        this.repository = repository;
+        this.webResourceRepository = webResourceRepository;
+    }
+
+    public ResponseEntity<ErrorMessage> upload(MultipartFile file, String id) {
+
+        if (file == null || file.isEmpty()) {
+            ErrorMessage errorMessage = new ErrorMessage("Argument", "Part named [file] is needed to successfully import a component");
+            logger.error(errorMessage.getMessage());
+            return new ResponseEntity<>(errorMessage, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        if (!MimeType.WEB_RESOURCES.matches(file.getContentType())) {
+            ErrorMessage errorMessage = new ErrorMessage("File content", "Only web resources files are allowed when importing a component");
+            logger.error(errorMessage.getMessage());
+            return new ResponseEntity<>(errorMessage, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        T component = repository.get(id);
+
+        WebResource<T> webResource = new WebResource<>();
+        webResource.setName(file.getName());
+        webResource.setMimeType(file.getContentType());
+        webResource.setComponent(component);
+
+        for (Iterator<WebResource<T>> webResourceIterator = component.getWebResources().iterator(); webResourceIterator.hasNext(); ) {
+            WebResource<T> existingWebResource = webResourceIterator.next();
+
+            //If the resource exist we delete the file before save the new one
+            if (webResource.equals(existingWebResource)) {
+                try {
+                    webResourceRepository.delete(webResource);
+                } catch (NotFoundException e) {
+                    logger.warn(String.format("Web resource to delete %s was not found", webResource.getName()), e);
+                } catch (IOException e) {
+                    ErrorMessage errorMessage = new ErrorMessage("Web resource deletion", String.format("Error while deleting webresource in %s [%s]", webResource.getName(), repository.getComponentName(), id));
+                    logger.error(errorMessage.getMessage(), e);
+                    return new ResponseEntity<>(errorMessage, HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+                webResourceIterator.remove();
```

Comment:
For this remove, no it's not dangerous to do that on an iterator
Java Iterators were created for this use case
http://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html